docs/README.md

```diff
@@ -0,0 +1,325 @@
+It is self-contained in a sense that a running EVE instance doesn't need to
+have access to any other information but this configuration object in order to
+perform tasks assigned to it by the controller.
+
+EVE depends on its configuration object so much so that it can not even control
+the hardware of the Edge Node (network interfaces, I/O ports, USB controllers,
+etc.) without retrieving hardware specific information from its configuration
+object. This model works very well from security and maintainability standpoint,
+but it presents a chicken-and-an-egg problem in one particular case: network
+interfaces. After all, EVE can only receive its configuration object from a
+Controller via network-based API call. This, in turn, requires at least a single
+functional hardware network interface and those can only be enabled after we
+receive configuration object from the Controller.
+
+EVE solves this problem by following a two-pronged approach:
+
+1. Having an out-of-band set of mechanisms that allow for the first
+   configuration object to be made available right after EVE instance boots
+2. Maintaining a sophisticated fallback semantics for the series of
+   configuration objects that EVE keeps receiving from the Controller
+
+The first part helps EVE bootstrap and configure itself during the initial boot
+sequence, the second part makes sure that EVE never ends up in a situation where
+the new configuration object made it impossible to receive further configuration
+objects due to erroneous configuration provided for network interfaces.
+
+There are currently two mechanisms for delivering out-of-band configuration
+object to EVE:
+
+1. Installing it during the [normal EVE installation process](#installing-eve-on-edge-nodes)
+2. Providing it to a running version of EVE on a specially formatted removable
+   media (USB stick, CDROM, external hard drive, etc.)
+
+Both methods start with obtaining a set of files (see the note below on how we
+are working towards making it a single file) and either putting them into the
+EVE configuration partition on the installation media (see [EVE Installation](#installing-eve-on-edge-nodes)
+for details) or using [mkusb.sh](../pkg/pillar/scripts/mkusb.sh) script to format
+removable media.
+
+It must be noted that currently we are still not quite there with out-of-band
+mechanism for delivery of EVE's configuration object. While ideal EVE
+implementation would simply be able to consume exactly the same protobuf encoded
+binary blob that it receives from the Controller, currently we still have to
+rely on an ad-hoc collection of configuration files that serve the same purpose.
+We expect these configuration files to go away relatively quickly, but for now
+EVE is still stuck with at least *DevicePortConfig/global.json* and
+*GlobalConfig/global.json* and they are documented in [legacy configuration](CONFIG.md).
+See [the following FAQ entry](FAQ.md) for how to manage both of these
+legacy files.
```

Comment:
This is important. The fact that we have a separate file documenting it, and not just buried in here, is a big deal. We may even need to highlight it elsewhere, but nicely done.